api/v1alpha1/workspace_validation.go

```diff
@@ -357,6 +406,32 @@ func (i *InferenceSpec) validateUpdate(old *InferenceSpec) (errs *apis.FieldErro
 	if (i.Template != nil && old.Template == nil) || (i.Template == nil && old.Template != nil) {
 		errs = errs.Also(apis.ErrGeneric("field cannot be unset/set if it was set/unset", "template"))
 	}
+	nameMap := make(map[string]string)
+	for _, adapter := range old.Adapters {
+		nameMap[adapter.Source.Name] = adapter.Source.Image
+	}
+
+	// check if adapter names are duplicate
+	for _, adapter := range i.Adapters {
+		errs = errs.Also(adapter.validateCreateorUpdate())
+	}
 
+	// check if adapter names are duplicate
+	if len(i.Adapters) > 0 {
+		errs = errs.Also(validateDuplicateName(i.Adapters, nameMap))
+	}
+	return errs
+}
+
+func validateDuplicateName(adapters []AdapterSpec, nameMap map[string]string) (errs *apis.FieldError) {
+	for _, adapter := range adapters {
+		if previousAdapterImage, ok := nameMap[adapter.Source.Name]; ok {
+			if previousAdapterImage != adapter.Source.Image {
```

Comment:
Why do you check image here? We only need to make sure the names are not duplicated.